test/nuget-parser.test.ts

```diff
@@ -0,0 +1,59 @@
+import * as tap from 'tap';
+const test = tap.test;
+import { getTargetFrameworkFromPackagesConfig } from '../lib/nuget-parser';
+
+test('various error handling is performed for getTargetFrameworkFromPackagesConfig', async (t) => {
+  // give bad content and expect to throw
+  const malformedXml = '<hello></bye>';
+  try {
+    await getTargetFrameworkFromPackagesConfig(malformedXml);
+    t.fail('expected to throw on malformed xml');
+  } catch (error) {
+    t.pass('expected to throw on malformed xml');
+  }
+  
+  // give empty content and expect undefined
+  try {
+    const result = await getTargetFrameworkFromPackagesConfig('');
+    t.pass('expected not to throw on empty content');
+    t.deepEqual(result, undefined, 'should return undefined on empty content');
+  } catch (error) {
+    t.fail('expected not to throw on empty content');
+  }
+
+  // give no packages but don't expect to throw
+  const noPackages = `
+    <?xml version="1.0" encoding="utf-8"?> 
+  `;
+  try {
+    const result = await getTargetFrameworkFromPackagesConfig(noPackages);
+    t.pass('expected not to throw on missing packages element in the xml');
+    t.deepEqual(result, null, 'should return undefined on missing packages element');
```

Comment:
This should compare to `undefined`, not sure why tap didn't catch this considering it's a "deep equal"